include/seqan3/argument_parser/detail/format_ctd.hpp

```diff
@@ -0,0 +1,484 @@
+     * \param[in]  long_id   The long identifier for the option (e.g. "integer").
+     * \param[in]  desc      The description of the option.
+     * \param[in]  spec      Advanced option specification. see seqan3::option_spec.
+     * \param[in]  validator The validator applied to the value after parsing (callable).
+     */
+    template<typename option_type, typename validator_type>
+    void add_option(option_type & value,
+                    char const short_id,
+                    std::string const & long_id,
+                    std::string const & desc,
+                    option_spec const & spec,
+                    validator_type && validator) 
+    {
+        // TODO (emanueleparisi) The current version of the CTD exporter does not support list options.
+        if (SequenceContainer<option_type> && 
+            !std::is_same_v<option_type,
+                            std::string>)
+        {
+            throw parser_design_error("At the moment, the CTD exporter does not support list options");
+        }
+        
+        // Do not report in the CTD file options marked as HIDDEN.
+        if (spec == HIDDEN)
+        {
+            return;
+        }
+
+        // Register 'clielement' node generation callback.
+        clielement_option_callbacks.push_back([this,
+                                               short_id,
+                                               long_id] (std::string app_name) {
+            if (long_id.empty())
+            {
+                append_ctd_clielement_node(prepend_dash(short_id), 
+                                           build_reference_name(app_name,
+                                                                short_id));
+            }
+            else
+            {
+                append_ctd_clielement_node(prepend_dash(long_id), 
+                                           build_reference_name(app_name,
+                                                                long_id));
+            }
+        });
+
+        // Register ITEM callback.
+        item_option_callbacks.push_back([this,
+                                         value,
+                                         short_id,
+                                         long_id,
+                                         desc,
+                                         spec,
+                                         validator] () {
+            std::string argument_name = {};
+            std::string argument_required = {};
+            std::string argument_advanced = {};
+
+            // Get correct value for argument name and required/advanced flags.
+            if (long_id.empty())
+            {
+                argument_name = std::string{short_id};
+            }
+            else
+            {
+                argument_name = long_id;
+            }
+            if (spec == REQUIRED)
+            {
+                argument_required = "true";
```

Comment:
initialise the values argument_required etc with "false" above and set only the one to "true" that is selected. What happens if both are true. I guess it should not be allowed since advanced options are not seen by the user by default. In this case yous should throw parse_design_error.